File: tests/fixtures/typescript_basic/src/widget.test.ts
// Test file for the TypeScript scan fixture.
import { Widget, WidgetKind, helper, answer } from "./widget";

function testWidgetRuns() {
  const w = new Widget(1);
  if (w.run() !== 1) throw new Error("expected 1");
}

function testAnswer() {
  if (answer() !== 42) throw new Error("expected 42");
}

function testHelper() {
  if (helper(5) !== 5) throw new Error("expected 5");
}

File: tests/fixtures/go_basic/widget/widget_test.go
package widget

import "testing"

// TestAnswer exercises Answer.
func TestAnswer(t *testing.T) {
	if Answer() == "" {
		t.Fatal("expected non-empty answer")
	}
}

// TestHelper exercises helper.
func TestHelper(t *testing.T) {
	if helper() != NAME {
		t.Fatalf("expected %q", NAME)
	}
}
